Assets/Colyseus/Runtime/Scripts/Models/ColyseusClient.cs

```diff
@@ -304,29 +311,57 @@ public async Task<T[]> GetAvailableRooms<T>(string roomName = "", Dictionary<str
                 queryString.Add("reconnectionToken", room.ReconnectionToken);
             }
 
-            room.SetConnection(CreateConnection(response.room, queryString, headers));
+            ColyseusRoom<T> targetRoom = previousRoom ?? room;
+            if (response.devMode) // Integrate devMode server re-connection callback
+            {
+	            room.SetConnection(CreateConnection(response.room, queryString, headers), async () =>
+	            {
+		            LSLog.Log($"[Colyseus devMode]: Re-establishing connection with room id {room.RoomId}");
```

Comment:
need to keep track of the number of retries within this scope - i suggest creating a local `Action` here, similarly to how this is approached in the javascript implementation.